test_app/tests/resource_registry/test_workload_identity_client.py

```
@@ -138,53 +162,91 @@ def test_service_token_property_auto_refresh(self, mock_get_service_token):
+        json_body = mock_request.call_args[1]["json"]
+        if expected_in_json is None:
+            assert "workload_ttl_seconds" not in json_body
+        else:
+            assert json_body["workload_ttl_seconds"] == expected_in_json
+
+    @pytest.mark.parametrize("invalid_ttl", [0, -1, -3600])
+    def test_request_workload_jwt_rejects_invalid_ttl(self, invalid_ttl):
+        """Test that workload_ttl_seconds=0 or negative raises ValueError (Gateway rejects 0)."""
+        client = WorkloadIdentityClient(base_url="https://gateway.example.com")
+
+        with pytest.raises(ValueError, match="must be None.*or >= 1"):
```

Ruff (0.15.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)